src/libcore/iter/adapters/mod.rs

```diff
@@ -252,11 +252,12 @@ where
 {
 }
 
+// FIXME(matthewjasper) This impl is much less general than it could be so that
+// it can easily be checked as being always applicable.
 #[doc(hidden)]
-unsafe impl<'a, I, T: 'a> TrustedRandomAccess for Copied<I>
```

Comment:
I'm not sure why this implementation is removed? There is no specialization involved here and this implementation is always applicable (all those bounds are required by `Iterator`'s `copied` method).

Reply:
TrustedRandomAccess is used to specialize the implementation of `ZipImpl`, so implementing it is essentially a specialization. The bounds on `copied` don't really help because of subtyping. I've removed this from this PR while I try to work around this issue.